raiden/network/proxies/token_network.py

```diff
@@ -0,0 +1,773 @@
+        """ Returns true if the channel is in a closed state, false otherwise. """
+        channel_data = self.detail_channel(partner)
+
+        if not isinstance(channel_data['state'], typing.T_ChannelState):
+            raise ValueError('channel state must be of type ChannelState')
+
+        return channel_data.get('state') == CHANNEL_STATE_SETTLED
+
+    def closing_address(self, partner: typing.Address):
+        """ Returns the address of the closer, if the channel is closed, None
+        otherwise. """
+        participants_data = self.detail_participants(partner)
+
+        if participants_data.get('our_is_closer'):
+            return participants_data.get('our_address')
+        elif participants_data.get('partner_is_closer'):
+            return participants_data.get('partner_address')
+
+        return None
+
+    def can_transfer(self, partner: typing.Address):
+        """ Returns True if the channel is opened and the node has deposit in
+        it.
+
+        Note: Having a deposit does not imply having a balance for off-chain
+        transfers. """
+        closed = self.channel_is_closed(partner)
+
+        if closed is True:
+            return False
+
+        return self._detail_participant(self.node_address, partner)['deposit'] > 0
+
+    def deposit(self, total_deposit, partner: typing.Address):
+        """ Set total token deposit in the channel to total_deposit.
+
+        Raises:
+            ChannelBusyError: If the channel is busy with another operation
+            RuntimeError: If the token address is empty.
+        """
+        if not isinstance(total_deposit, int):
+            raise ValueError('total_deposit needs to be an integral number.')
+
+        token_address = self.token_address()
+
+        token = Token(
+            self.client,
+            token_address,
+            self.poll_timeout,
+        )
+        current_balance = token.balance_of(self.node_address)
+        current_deposit = self._detail_participant(self.node_address, partner)['deposit']
+        amount_to_deposit = total_deposit - current_deposit
+
+        if amount_to_deposit <= 0:
+            raise ValueError('deposit [{}] must be greater than 0.'.format(
+                amount_to_deposit,
+            ))
+
+        if current_balance < amount_to_deposit:
+            raise ValueError('deposit [{}] cant be larger than the available balance [{}].'.format(
+                amount_to_deposit,
+                current_balance,
+            ))
+
+        log.info(
+            'deposit called',
+            token_network=pex(self.address),
+            node=pex(self.node_address),
+            partner=pex(partner),
+            total_deposit=total_deposit,
+            amount_to_deposit=amount_to_deposit,
+        )
+
+        if not self.channel_operations_lock.acquire(blocking=False):
+            raise ChannelBusyError(
+                f'Channel with address {self.address} is '
```

Comment:
This won't work. Address should be given with `pex()` but also we no longer have channel addresses. Use the channel identifier instead.

Reply:
Used the two participant addresses for now. For the `channel_identifier` I would need to do another contract call to get it, unless we use an internal function to compute it. But I think using the addresses should be ok and provides more context.